sbi/inference/posteriors/base_posterior.py

```diff
@@ -355,26 +359,42 @@ def _slice_np_mcmc(
             initial_params: Initial parameters for MCMC chain.
             thin: Thinning (subsampling) factor.
             warmup_steps: Initial number of samples to discard.
+            vectorized: Whether to use a vectorized implementation of
+                the Slice sampler (still experimental).
+            show_progress_bars: Whether to show a progressbar during sampling;
+                can only be turned off for vectorized sampler.
 
         Returns: Tensor of shape (num_samples, shape_of_single_theta).
         """
-
         num_chains = initial_params.shape[0]
         dim_samples = initial_params.shape[1]
 
-        all_samples = []
-        for c in range(num_chains):
-            posterior_sampler = SliceSampler(
-                utils.tensor2numpy(initial_params[c, :]).reshape(-1),
-                lp_f=potential_function,
-                thin=thin,
+        if not vectorized:  # Sample all chains sequentially
+            all_samples = []
+            for c in range(num_chains):
+                posterior_sampler = SliceSampler(
+                    utils.tensor2numpy(initial_params[c, :]).reshape(-1),
+                    lp_f=potential_function,
+                    thin=thin,
+                )
+                if warmup_steps > 0:
+                    posterior_sampler.gen(int(warmup_steps))
+                all_samples.append(posterior_sampler.gen(int(num_samples / num_chains)))
```

Comment:
Shouldn't we use `ceil()` here? If `num_samples` is not a multiple of `num_chains`, it will not produce enough samples, no?